src/qt/overviewpage.cpp

```diff
@@ -398,8 +403,25 @@ void OverviewPage::onRefreshClicked()
 
 void OverviewPage::migrateClicked()
 {
-    if(walletModel->getAvailableLelantusCoins() && spark::IsSparkAllowed() && chainActive.Height() < ::Params().GetConsensus().nLelantusGracefulPeriod){
-        MigrateLelantusToSparkDialog migrate(walletModel);
+    WalletModel::EncryptionStatus status = walletModel->getEncryptionStatus();
+    auto privateBalance = walletModel->getLelantusModel()->getPrivateBalance();
+    auto lGracefulPeriod = ::Params().GetConsensus().nLelantusGracefulPeriod;
+    migrateAmount = "<b>" + BitcoinUnits::formatHtmlWithUnit(walletModel->getOptionsModel()->getDisplayUnit(), privateBalance.first);
+    migrateAmount.append("</b>");
+    QString info = tr("Your wallet needs to be unlocked to migrate your funds to Spark.");
+
+    if(status == WalletModel::Locked) {
+
+        AskPassphraseDialog dlg(AskPassphraseDialog::Unlock, this, info);
+        dlg.setModel(walletModel);
+        dlg.exec();
+        ui->textWarning2->setText(tr("To complete the migration of %1").arg(migrateAmount));
+        ui->textWarning1->hide();
+        ui->warningFrame->show();
+    } else {
```

Comment:
Your logic will open the wallet unlock window, but after unlock you will need to click again to do the migration, I think we need to do the following logic,
- if wallet is locked
    Show the window and ask to unlock
- if wallet is not locked or it is unlocked by upper step, so the migration